traits/tests/test_trait_set_object.py

```diff
@@ -0,0 +1,515 @@
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import pickle
+import unittest
+from unittest import mock
+
+from traits.api import HasTraits, Set, Str
+from traits.trait_errors import TraitError
+from traits.trait_set_object import TraitSet, adapt_trait_validator
+from traits.trait_types import _validate_int
+
+
+def int_validator(current_set, value):
+    return {_validate_int(v) for v in value}
+
+
+def string_validator(current_set, value):
+    ret = set()
+    for v in value:
+        if isinstance(v, str):
+            ret.add(v)
+        else:
+            raise TraitError
+    return ret
+
+
+class TestTraitSet(unittest.TestCase):
+
+    def setUp(self):
+        self.added = None
+        self.removed = None
+        self.validator_args = None
+
+    def notification_handler(self, removed, added):
+        self.removed = removed
+        self.added = added
+
+    def validator(self, set_, added):
+        self.validator_args = (set_, added)
+        return added
+
+    def test_init(self):
+        ts = TraitSet({1, 2, 3})
+
+        self.assertSetEqual(ts, {1, 2, 3})
+        self.assertIsNone(ts.validator)
+        self.assertEqual(ts.notifiers, [])
+
+    def test_init_with_no_input(self):
+        ts = TraitSet()
+
+        self.assertSetEqual(ts, set())
+        self.assertIsNone(ts.validator)
+        self.assertEqual(ts.notifiers, [])
+
+    def test_validator(self):
+        ts = TraitSet({1, 2, 3}, validator=int_validator)
+
+        self.assertSetEqual(ts, {1, 2, 3})
+        self.assertEqual(ts.validator, int_validator)
+        self.assertEqual(ts.notifiers, [])
+
+    def test_adapt_trait_validator(self):
+
+        def bool_validator(object, name, value):
+            if isinstance(value, bool):
+                return value
+            else:
+                raise TraitError
+
+        # Fail without adaptor
+        with self.assertRaises(TypeError):
+            TraitSet({}, validator=bool_validator)
+
+        # Attach the adaptor
+        set_bool_validator = adapt_trait_validator(bool_validator)
+
+        # It now works!
+        ts_2 = TraitSet({}, validator=set_bool_validator)
+        ts_2.update([True, False, True])
+
+        # Decorate with set adaptor
+        @adapt_trait_validator
+        def bool_validator(object, name, value):
+            if isinstance(value, bool):
+                return value
+            else:
+                raise TraitError
+
+        # Still working
+        ts = TraitSet({}, validator=bool_validator)
+        ts.update([True, False, True])
+
+        with self.assertRaises(TraitError):
+            ts.update(["invalid"])
+
+    def test_notification(self):
+        ts = TraitSet({1, 2, 3}, validator=int_validator,
+                      notifiers=[self.notification_handler])
+
+        self.assertSetEqual(ts, {1, 2, 3})
```

Comment:
I don't think there's any reason to use `assertSetEqual`; just use `assertEqual` throughout. As the [documentation](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addTypeEqualityFunc) says: "it’s usually not necessary to invoke these methods directly".